### ISSUE TITLE
Please include NATS Tools in docker container

### ISSUE THREAD

**Issue by ryn9** [nats-io/nats-docker#143]
It would be helpful to have the nats and nats-top commands available in the container.

That way, we could execute the following to interface with the running instance:
docker exec -it CONTAINER_NAME nats
docker exec -it CONTAINER_NAME nats-top

references:
https://docs.nats.io/using-nats/nats-tools/nats_cli
https://docs.nats.io/using-nats/nats-tools/nats_top

**Comment by jnehlmeier** [comment]
There is https://hub.docker.com/r/natsio/nats-box which contains the tools.

**Comment by ryn9** [comment]
hmm - didnt see that.
Maybe reference that in the readme of this effort?

**Comment by jnehlmeier** [comment]
Indeed it is a bit hard to discover. I stumbled upon it while reading https://docs.nats.io/running-a-nats-service/nats_docker#testing-the-clusters

I think there should be an additional docker section/page in https://docs.nats.io/using-nats/nats-tools to mention it.

**Comment by kostyaten** [comment]
This does not solve the problem when you need to check if the service is running. Or can you tell me how to check?
My example is docker-compose.yaml

`
    healthcheck:
      test: "how?????"
      interval: 10s
      timeout: 5s
      retries: 5
`

**Comment by jnehlmeier** [comment]
You can turn on monitoring (https://docs.nats.io/running-a-nats-service/configuration/monitoring) and then use curl to call the healthz endpoint.

All endpoints: https://docs.nats.io/running-a-nats-service/nats_admin/monitoring

**Comment by kostyaten** [comment]
In order to use it, curl, wget or nc must be available inside the container. But this is not available inside your image.
see https://docs.docker.com/compose/how-tos/startup-order/ and https://docs.docker.com/reference/compose-file/services/#healthcheck

**Comment by jnehlmeier** [comment]
@kostyaten Yes. For the time being you would need to create your own image, extending from nats:alpine and then you can install curl using apk in your Dockerfile.

**Comment by Roman-Weisman** [comment]
Not having the curl in the image or an easy healthcheck is frustrating.
Also - tried to set the nats_version to latest by adding jq to apk, and extracting from json, but it doesn't work from the dockerfile, it does from local though:
curl -s https://api.github.com/repos/nats-io/nats-server/releases/latest | jq -r '.tag_name' | sed 's/^v//' 

Anyhow, for others facing the same issue an need a healthcheck:

yml:
services:
  nats:
    build:
      context: .
      dockerfile: Dockerfile.nats
    container_name: nats-server
    healthcheck:
      test: ["CMD-SHELL", "curl -f http://localhost:8222/healthz || exit 1"]
      interval: 10s
      timeout: 5s
      retries: 3
      start_period: 5s

Dockerfile.nats:
# Nats minimal image is missing curl - wihtout it the healthcheck fails, that's why alpine wrapper with curl is required.
FROM alpine
RUN apk add --no-cache curl unzip

# Download and install NATS server
ENV NATS_VERSION=2.10.11
RUN curl -L https://github.com/nats-io/nats-server/releases/download/v${NATS_VERSION}/nats-server-v${NATS_VERSION}-linux-amd64.zip -o nats.zip && \
    unzip nats.zip && \
    mv nats-server-v${NATS_VERSION}-linux-amd64/nats-server /usr/local/bin/nats-server && \
    chmod +x /usr/local/bin/nats-server && \
    rm -rf nats.zip nats-server-v${NATS_VERSION}-linux-amd64

# Run NATS with monitoring enabled
CMD ["nats-server", "-m", "8222"]

**Comment by Anthonyntilelli** [comment]
+1 to this ticket, heath check for NATS would be a great addition.
